motionplan/tpspace/ptgIK.go

```diff
@@ -100,32 +100,29 @@ func NewPTGIK(simPTG PTG, logger logging.Logger, refDistLong, refDistShort float
 func (ptg *ptgIK) Solve(
 	ctx context.Context,
 	solutionChan chan<- *ik.Solution,
-	seed []referenceframe.Input,
+	seedVals []referenceframe.Input,
 	solveMetric ik.StateMetric,
 	nloptSeed int,
 ) error {
 	internalSolutionGen := make(chan *ik.Solution, 1)
 	defer close(internalSolutionGen)
 	var solved *ik.Solution
-	var gridSolved *ik.Solution
 
-	if seed == nil {
-		seed = ptg.defaultSeed
+	seed := make([]referenceframe.Input, 0, len(ptg.defaultSeed))
+	for _, defaultSeedVal := range ptg.defaultSeed {
+		seed = append(seed, referenceframe.Input{defaultSeedVal.Value})
 	}
 
-	err := ptg.gridSim.Solve(ctx, internalSolutionGen, seed, solveMetric, nloptSeed)
-	if err != nil {
-		return err
-	}
-	select {
-	case gridSolved = <-internalSolutionGen:
-	default:
+	if seedVals != nil {
+		// use the passed seed as the first dist if in range
+		if seedVals[0].Value < ptg.ptgFrame.DoF()[1].Max {
```

Comment:
I'd prefer to move this to the caller code and simply use the seedVal as passed in

Reply:
Done